### ISSUE TITLE
Are the caches working?

### ISSUE THREAD

**Issue by Elias-Graf** [lilyinstarlight/nixos-cosmic#301]
Are the caches working? Because every time I update, it seems like more and more packages are just build instead of downloaded. Or am I not supposed to run `nix flake update`? Do I need to do some step before then running `nixos-rebuild`?

Thanks in advance! :)

**Comment by Elias-Graf** [comment]
This is my flake, should be correct, as it used to work:

```nix
        perseus = nixpkgs.lib.nixosSystem {
          specialArgs = { inherit home-manager; };

          modules = [
            {
              nix.settings = {
                substituters = [ "https://cosmic.cachix.org/" ];
                trusted-public-keys = [ "cosmic.cachix.org-1:Dya9IyXD4xdBehWjrkPv6rtxpmMdRel02smYzA85dPE=" ];
              };
            }

            nixos-cosmic.nixosModules.default

            ./system/perseus.nix
            ./system/users/elias.nix

            home-manager.nixosModules.home-manager

            ./home/elias/perseus.nix
          ];
        };
```

And then in `system/perseus.nix`:

```nix
{ ... }:
{
  imports = [
    ./hardware-configuration/perseus.nix

    ./common/core

    # ./common/optional/desktop-environment/hyprland.nix
    ./common/optional/desktop-environment/cosmic.nix

    ./common/optional/fonts.nix
    ./common/optional/ssh.nix
  ];

  networking.hostName = "perseus";
}
```

and finishing with `system/common/optional/desktop-environment/cosmic.nix`:

```nix
{ pkgs, ... }: {
  services.desktopManager.cosmic.enable = true;
  services.displayManager.cosmic-greeter.enable = true;

  environment = {
    systemPackages = with pkgs; [ wl-clipboard ];
  };
}
```

**Comment by couchpotato007** [comment]
I have the same problem. After the updates yesterday, the cosmic desktop is no longer compiled. I am now using an older lock file until the problems are fixed. I was already wondering why it tries to compile all packages. I always get the following error message: 
“xdg-desktop-portal-cosmic> thread 'main' panicked at cargo-auditable/src/cargo_auditable.rs:40:39:
xdg-desktop-portal-cosmic> called Option::unwrap() on a None value
xdg-desktop-portal-cosmic> note: run with RUST_BACKTRACE=1 environment variable to display a backtrace”

**Comment by Elias-Graf** [comment]
That seems like an error, though. The compilation itself isn't successful. I'm pretty sure I can build manually, just takes ages.

**Comment by drakon64** [comment]
Cachix has a limit of 5GB per account. Once an account goes over this limit, derivations start to be deleted from the account.

This is _probably_ just an unfortunate case of the 5GB limit being exceeded by the upstream account and things getting deleted, and then end users having to compile packages because they're no longer in the binary cache.

**Comment by Elias-Graf** [comment]
Wouldn't that mean _old_ ones are getting deleted? Because I'm getting the issues with a brad new updated nix flake. For example, I'm compiling "cosmic-term-xxx-09-1". Or is it random? :D

Anyway can this me worked around somehow?

**Comment by lilyinstarlight** [comment]
It's more likely a staging-next iteration just hit nixos-unstable (or something that causes rebuilds) and this flake hasn't been built against it yet

**Comment by lilyinstarlight** [comment]
The cosmic package set and my personal OS repo still comes out to much less than 5 GiB so the latest can always be resident in the cache

**Comment by couchpotato007** [comment]
> That seems like an error, though. The compilation itself isn't successful. I'm pretty sure I can build manually, just takes ages.

Thanks for the tip about the job limit. At least it is now compiling again. Apparently my RAM overflowed which caused the error.

**Comment by Elias-Graf** [comment]
@lilyinstarlight Thanks for the information! Where and how can I track progress on this? It is simply up to this repo (probably you ^^) to build against the newest unstable version and push it to the cache?
I'm assuming I can't really help in any way except wait.

**Comment by Elias-Graf** [comment]
@couchpotato007 

> Thanks for the tip about the job limit. At least it is now compiling again. Apparently my RAM overflowed which caused the error.

There is some more information about that in the README.md. There is also `--cores`. But it's nice to hear you can build now,

**Comment by Elias-Graf** [comment]
I don't think it was last time I upgraded.

**Comment by Tommimon** [comment]
I'm facing the same issue, I'm using NixOS Stable and I have to compile all the Cosmic packages. I added

```
nix.settings = {
  substituters = [ "https://cosmic.cachix.org/" ];
  trusted-public-keys = [ "cosmic.cachix.org-1:Dya9IyXD4xdBehWjrkPv6rtxpmMdRel02smYzA85dPE=" ];
};
```

Any clue why it might not work?

**Comment by drakon64** [comment]
Did you run `nixos-rebuild test` before trying to enable the COSMIC modules?

**Comment by Tommimon** [comment]
> Did you run `nixos-rebuild test` before trying to enable the COSMIC modules?

My bad, I thought I could run it also afterwards, thanks a lot, I can confirm it works.
(This is a friendly reminder to follow instructions step by step)

**Comment by lilyinstarlight** [comment]
I've updated the example flake in the readme to use the nixpkgs rev from this flake, to avoid cache misses. This flake updates nixpkgs rev daily so it shouldn't be much of an issue for people waiting on nixpkgs bumps, but should avoid accidental attempts to use COSMIC packages not in the cache

If anyone is still having problems, please open a new issue. Thanks!

**Comment by Elias-Graf** [comment]
I seem to be experiencing cache misses again, can anyone confirm?

**Comment by Elias-Graf** [comment]
Seems to be working again since the update from yesterday evening (2024-11-18). Thank you very much!
